pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -471,7 +473,67 @@ cdef class _Timestamp(ABCTimestamp):
                                   field, freqstr, month_kw)
         return out[0]
 
-    cpdef _get_date_name_field(self, object field, object locale):
+    @property
+    def is_month_start(self) -> bool:
+        """
+        Return True if date is first day of month.
+        """
+        if self.freq is None:
```

Comment:
don't you now need to move freq to _Timestamp?

Reply:
dont _need_ to, no.  but i do want to.  saving that for next pass